src/t8_forest/t8_forest_iterate.h

```diff
@@ -44,20 +44,30 @@ typedef int (*t8_forest_iterate_face_fn) (t8_forest_t forest, t8_locidx_t ltreei
  * \param[in] leaf_elements       the leaf elements in \a forest that are descendants of \a element (or the element 
  *                                itself if \a is_leaf is true)
  * \param[in] tree_leaf_index     the local index of the first leaf in \a leaf_elements
- * \param[in] queries             if not NULL, a query that is passed through from the search function
- * \param[in] query_indices       if \a query is not NULL the indices of \a query in the \a queries array from \ref 
- *                                t8_forest_search
- * \param[in, out] query_matches  if \a query is not NULL: true at the i-th index if and only if the element 'matches' 
- *                                the query of the i-th query index. 
- *                                if \a query is NULL: true if and only if the search should continue with the
- *                                children of \a element and the queries should be performed for this element.
+ */
+typedef int (*t8_forest_search_fn) (t8_forest_t forest, const t8_locidx_t ltreeid, const t8_element_t *element,
+                                    const int is_leaf, const t8_element_array_t *leaf_elements,
+                                    const t8_locidx_t tree_leaf_index);
+
+/*
```

Comment:
The same as above: a few introductory lines about what this callback does.

Reply:
Also this should be a doxygen comment which it does not seem to be.
(Same above)